syntax = "proto3";

message Rule {
    RuleFilter filter = 1;
    RuleAction action = 2;
}

message RuleFilter {
    optional string vendor_regex = 1;
    optional string model_regex = 2;
}

enum RuleActionType {
    DROP_FILE = 0;
    MOVE_FILE = 1;
    DELETE_FILE = 2;
    COPY_DEVICE = 3;
}

message RuleAction {
    RuleActionType type = 1;
}

message RuleActionDropFile {
    RuleActionType type = 1;
    string src_path = 2;
    string dst_path = 3;
}

message RuleActionMoveFile {
    RuleActionType type = 1;
    string src_path = 2;
    string dst_path = 3;
}

message RuleActionDeleteFile {
    RuleActionType type = 1;
    string path = 2;
}
